### DISCUSSION TITLE
实现中文文献翻译为英文文献的引用？

### DISCUSSION THREAD

**Discussion by JamesSmith20** [zotero-chinese/wiki#328] [好点子 / Ideas]
请问有没有这么一种功能，把中文参考文献翻译为英文，并且在引用时标注(in Chinese)？如果不行，应该如何实现？非常感谢！

**Comment by zepinglee** [comment]
CSL 可以输出带英文翻译的格式，但是需要将英文翻译内容按照格式填写在 `extra` 字段。参考[《心理学报》样式](https://zotero-chinese.com/styles/acta-psychologica-sinica/)的说明，以及其中文[示例文献](https://www.zotero.org/groups/4677213/chinese_csl_development/collections/7SP7XX8X)。

**Reply by JamesSmith20** [reply]
我看了一下《心理学报》样式，认为这个就是我想要的，非常感谢！但我发现《心理学报》样式并未实现英文的（作者，出版年）的作者项的英文化，请问一下应该如何处理，以使得（作者，出版年）可以变成(Author, Year) ？

**Reply by JamesSmith20** [reply]
我在https://github.com/l0o0/translators_CN/issues/257 发现了心理学报original-author的正确填写方式，非常感谢。
我现在有两个问题。
首先，我不知道能否在抓取时直接获得中文论文、杂志名的英文翻译。
其次，我想在正文中将(作者,Year)改为(Author, Year)。
我可以用海外VPN访问知网oversea版本，获得中文论文、杂志名的英文翻译。同时用Excel的VBA来获得original-author，但这样比较费时费力。

original-author: Zhang || San
original-author: Li || Si
original-title: The past and future of Chinese psychology
original-container-title: Acta Psychologica Sinica

**Reply by JamesSmith20** [reply]
分割名字主要是把"Zhang San; LI Si;"或者"ZHANG San; Li Si"改为
original-author: Zhang || San<br>original-author: Li || Si<br>
规律是前面加上original-author: ；同时把姓名分割为"姓 || 名"，同时确保只有首字母大写。分割名字的VBA代码如下：

```
Function FormatAPA7ToOriginalAuthor(NameList As String) As String
    Dim Names() As String
    Dim FormattedNames As String
    Dim i As Integer
    Dim FullName() As String
    Dim Surname As String
    Dim GivenName As String
    
    On Error GoTo ErrorHandler
    
    ' 分割名字列表
    Names = Split(NameList, ";")
    
    For i = LBound(Names) To UBound(Names)
        If Trim(Names(i)) <> "" Then
            ' 分割姓和名
            FullName = Split(Trim(Names(i)), " ")
            
            ' 确保名字至少包含姓和名
            If UBound(FullName) > 0 Then
                Surname = Application.Proper(FullName(0)) ' 姓的首字母大写
                GivenName = Application.Proper(FullName(1)) ' 名的首字母大写
                
                ' 拼接格式为 original-author: 姓 || 名
                FormattedNames = FormattedNames & "original-author: " & Surname & " || " & GivenName & "<br>"
            Else
                ' 单字处理（假设为姓）
                Surname = Application.Proper(FullName(0))
                FormattedNames = FormattedNames & "original-author: " & Surname & " || <br>"
            End If
        End If
    Next i
    
    ' 返回结果
    FormatAPA7ToOriginalAuthor = Trim(FormattedNames)
    Exit Function

ErrorHandler:
    FormatAPA7ToOriginalAuthor = "#ERROR!"
End Function

```

**Reply by zepinglee** [reply]
> 但我发现《心理学报》样式并未实现英文的（作者，出版年）的作者项的英文化，请问一下应该如何处理，以使得（作者，出版年）可以变成(Author, Year) ？

@JamesSmith20 因为《心理学报》这么规定的。你可以将这里改为 `original-author`。

https://github.com/zotero-chinese/styles/blob/6cf0326a9aada693269ae860bb133523bd9b6a95/src/acta-psychologica-sinica/acta-psychologica-sinica.csl#L602

**Reply by JamesSmith20** [reply]
非常感谢
